addons/calendar/models/calendar_alarm_manager.py

```diff
@@ -159,19 +160,25 @@ def _get_events_to_notify(self, ttype):
                AND "event"."start" - CAST("alarm"."duration" || ' ' || "alarm"."interval" AS Interval) < now() at time zone 'utc'
              )''', [ttype, self.env.context['lastcall']])
 
-        ids = [row[0] for row in self.env.cr.fetchall()]
-        return self.env['calendar.event'].browse(ids)
+        results = self.env.cr.fetchall()
+        alarms_by_event = dict({result[0]: [result[1]] if isinstance(result[1], int) else result[1] for result in results}) if len(results) else False
```

Comment:
As you're doing a join (no outer join I mean), how can result[1] (=alarm.id) be of another type than int?